streams/src/main/java/org/apache/kafka/streams/kstream/TimeWindowedDeserializer.java

```diff
@@ -33,22 +33,22 @@
  */
 public class TimeWindowedDeserializer<T> implements Deserializer<Windowed<T>> {
 
-    private final Long windowSize;
+    private Long windowSize;
     private boolean isChangelogTopic;
 
     private Deserializer<T> inner;
-    
+
     // Default constructor needed by Kafka
     public TimeWindowedDeserializer() {
-        this(null, Long.MAX_VALUE);
+        this(null, null);
     }
 
-    // TODO: fix this part as last bits of KAFKA-4468
+    @Deprecated
     public TimeWindowedDeserializer(final Deserializer<T> inner) {
         this(inner, Long.MAX_VALUE);
```

Comment:
Do you want to change this from `Long.MAX_VALUE` as well?

Reply:
I thought about it but figured since it's deprecated anyway and we want to keep backwards compatibility I would leave it. WDYT?

Reply:
You are probably right since we are deprecating it. But I don't have a strong opinion on this either way